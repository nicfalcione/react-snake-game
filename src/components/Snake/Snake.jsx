import React from "react";
import { useSelector } from "react-redux";
import SnakeSegment from '../SnakeSegment/SnakeSegment';
import './Snake.css';

const Snake = () => {
    const snakeSegments = useSelector((state) => state.snakeSegments);

    return (
        <>
            {
                snakeSegments.map(segment =>
                    <SnakeSegment key={segment.key}
                                  xPos={segment.xPos}
                                  yPos={segment.yPos}
                    /> 
                )
            }
        </>
    );
}

export default Snake;