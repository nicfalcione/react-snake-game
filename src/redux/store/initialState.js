import { RIGHT } from '../../constants/constants';

const defaultState = {
    isRunning: true,
    gameOver: false,
    direction: RIGHT,
    score: 0,
    speed: 100,
    snakeSegments:[
        {
            key: 1,
            xPos: 380,
            yPos: 220
        },
        {
            key: 2,
            xPos: 360,
            yPos: 220
        },
        {
            key: 3,
            xPos: 340,
            yPos: 220
        }
    ],
    apple: {
        xPos: 400,
        yPos: 400
    }
};

export default defaultState;