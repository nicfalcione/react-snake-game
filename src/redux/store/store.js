import { createStore } from '@reduxjs/toolkit';
import { gameStateReducer } from '../reducers/gameStateReducer';

const store = createStore(gameStateReducer);

export default store;