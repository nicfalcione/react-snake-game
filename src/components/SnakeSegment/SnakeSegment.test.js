import { render } from "@testing-library/react";
import React from "react";
import { Provider } from "react-redux";
import store from '../../redux/store/store';
import Snake from '../Snake/Snake';

const setupRender = () => {
    return render(
        <Provider store={store}><Snake/></Provider>
    );
}

describe('SnakeSegment component tests', () => {
    it('should match snapshot', () => {
        const { container } = setupRender();
        expect(container).toMatchSnapshot();
    });
});