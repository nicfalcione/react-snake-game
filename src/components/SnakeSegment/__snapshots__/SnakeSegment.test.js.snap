// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SnakeSegment component tests should match snapshot 1`] = `
<div>
  <div
    class="game-snake-segment"
    style="left: 380px; top: 220px;"
  />
  <div
    class="game-snake-segment"
    style="left: 360px; top: 220px;"
  />
  <div
    class="game-snake-segment"
    style="left: 340px; top: 220px;"
  />
</div>
`;
